From 19e7b940e1205c827a553395ea6e3e19146124a9 Mon Sep 17 00:00:00 2001
From: Sven Neumann <s.neumann@raumfeld.com>
Date: Tue, 17 May 2011 13:00:22 +0200
Subject: [PATCH 2/2] Don't unref cancelled SoupMessage

The SoupMessage that is cancelled in the dispose handler doesn't
belong us, it has already been queued. This fixes a crash that
occured when a GUPnPService object is disposed before the initial
introspection call has finished.
---
 libgupnp/gupnp-service-info.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/libgupnp/gupnp-service-info.c b/libgupnp/gupnp-service-info.c
index 6ce9c44..c807c2d 100644
--- a/libgupnp/gupnp-service-info.c
+++ b/libgupnp/gupnp-service-info.c
@@ -195,7 +195,6 @@ gupnp_service_info_dispose (GObject *object)
                                                      data->message,
                                                      SOUP_STATUS_CANCELLED);
 
-                        g_object_unref (data->message);
                         get_scpd_url_data_free (data);
 
                         info->priv->pending_gets =
-- 
1.7.1

